### PR TITLE
Update the custom processor project

### PR DESCRIPTION
**Pull request by reyang**
Trying to use the docs and tutorial projects to drive for better design and implementation.
This change should give us some idea about the processor use case.
I plan to work on the sampler and exporter tutorial projects in the next few PRs, then we can use these projects to question ourselves "are these the developer experience we want and are there places we can improve" - similar like what we did [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/988#discussion_r464778565).

## Changes

Added multiple processors demo.

### CODE REVIEW

File: docs/trace/building-your-own-processor/Program.cs [hunk 1]
```diff
@@ -27,10 +27,12 @@ public static void Main()
...
         using var otel = Sdk.CreateTracerProvider(b => b
             .AddActivitySource("MyCompany.MyProduct.MyLibrary")
-            .AddProcessorPipeline(pipeline =>
-            {
-                pipeline.AddProcessor(current => new MyActivityProcessor());
-            }));
+
+            // TODO: seems buggy if you remove A and B here, MyActivityProcessor(C).OnEnd is not called.
+            // TODO: should the dispose order be C, B, A or A, B C?
+            .AddProcessorPipeline(p => p.AddProcessor(current => new MyActivityProcessor("A")))
```
Comment (cijothomas) [inline docs/trace/building-your-own-processor/Program.cs:33]: This is adding multiple processor pipelines. I guess you wanted to add multiple processors to the same, single pipeline?

File: docs/trace/building-your-own-processor/MyActivityProcessor.cs [hunk 2]
```diff
@@ -22,29 +22,37 @@
 
 internal class MyActivityProcessor : ActivityProcessor
 {
-    public MyActivityProcessor()
+    private readonly string name;
+
+    public MyActivityProcessor(string name)
```
Comment (cijothomas) [inline docs/trace/building-your-own-processor/MyActivityProcessor.cs:27]: Need to take the next ActivityProcessor, and call its corresponding methods in Start/End/etc.

Comment (reyang) [reply docs/trace/building-your-own-processor/MyActivityProcessor.cs:27]: For demonstration I was trying to make it simple by not chaining the processors at all.

File: docs/trace/building-your-own-processor/Program.cs (hunk 1 above)
Comment (reyang) [reply docs/trace/building-your-own-processor/Program.cs:33]: Nope, I was using this to showcase how pipeline could be confusing, and I think we need to change it.

Comment (reyang) [reply docs/trace/building-your-own-processor/Program.cs:33]: The fan out scenario is a separate topic, and the name should be changed #979.

**Pull request merged**